File: src/ast/ThrowExpression.java
package ast;

import java.io.StringWriter;

public class ThrowExpression extends BaseNode {
    private BaseNode _expression;

    public ThrowExpression(BaseNode expression) {
        super(NodeType.ThrowExpression);
        _expression = expression;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("throw ");
        _expression.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/ThrowExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ThrowExpression(private val expression: BaseNode) : BaseNode(NodeType.ThrowExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("throw ")
        expression.print(writer)
    }

}